hasher-matcher-actioner/docker-compose.prod.yaml

```diff
@@ -0,0 +1,71 @@
+version: "3"
+
+x-postgres-variables: &postgres-variables
+  POSTGRES_USER: media_match
+  # TODO: Use a secret instead.
+  POSTGRES_PASSWORD: hunter2
+  POSTGRESS_HOST: db
+  POSTGRESS_DBNAME: media_match
+
+services:
+  indexer:
+    build: .
+    command: ["gunicorn", "--bind", "0.0.0.0:8080", "OpenMediaMatch.app:create_app()", "--log-level", "debug", "-w", "1"]
+    restart: unless-stopped
+    ports:
+      - 8081:8080
+    environment:
+      OMM_CONFIG: /build/omm_config.py
+      PRODUCTION: "true"
+      ROLE_HASHER: "false"
+      ROLE_MATCHER: "false"
+      ROLE_CURATOR: "false"
+      TASK_INDEXER: "true"
```

Comment:
blocking: This feels like it's heading in the wrong direction for the default setup, and we might be in the territory of something that might be deployment-specific.

Instead, here's what I propose we do:
1. Land #1574 to create more reference configs by base
2. Add support for using environment variables via https://flask.palletsprojects.com/en/2.3.x/config/#configuring-from-environment-variables
3. In the default docker-compose in this file, we can either pick a worker role by default or another prod=True setup
4. In your own deployment in your infra, you should not use this docker file, and instead have your own.